addon/components/tri-state.js

```diff
@@ -1,281 +1,193 @@
 
     /**
-     * Reject request immediately if any of the promises are rejected.
-     * @type {Boolean}
+     * One or more actions that each return a promise.
+     * @type {Promise|Object|Array}
      */
-    this.failFast = this.getWithDefault('failFast', true);
+    this.promises = this.get('promises');
 
     /**
      * If true, display the last successful data instead of the loading or error state when
      * fetching new data.
      * @type {Boolean}
+     * @default false
      */
     this.showLastSuccessful = this.getWithDefault('showLastSuccessful', false);
 
     /**
+     * Fulfill `promise`, even if one or more batched `promises` fail
+     * @type {Boolean}
+     * @default false
+     */
+    this.forceResolveAll = this.getWithDefault('forceResolveAll', false);
+
+    /**
      * Name of the component to render when any state is inactive.
      * @type {String}
+     * @default 'tri-noop'
      */
     this.noopComponent = this.getWithDefault('noopComponent', 'tri-noop');
 
     /**
      * Name of the component to render when any state is active, unless overridden.
      * @type {String}
+     * @default 'tri-yield'
      */
     this.yieldComponent = this.getWithDefault('yieldComponent', 'tri-yield');
 
     /**
      * Name of the component to render when the error state is active.
      * @type {String}
+     * @default 'tri-yield'
      */
     this.errorComponent = this.getWithDefault('errorComponent', this.yieldComponent);
 
     /**
      * Name of the component to render when the success state is active.
      * @type {String}
+     * @default 'tri-yield'
      */
     this.successComponent = this.getWithDefault('successComponent', this.yieldComponent);
 
     /**
      * Name of the component to render when the loading state is active.
      * @type {String}
+     * @default 'tri-yield'
      */
     this.loadingComponent = this.getWithDefault('loadingComponent', this.yieldComponent);
 
     /**
-     * Whether or not we should set an event listener so outside components can trigger data fetches
-     * @type {Boolean}
+     * Callback called when the promise has successfully resolved
+     * @type {Function}
+     * @default noop
      */
-    this.listenForEvents = this.getWithDefault('listenForEvents', false);
-
-    // If we want to allow for outside sources to take action, set up event listeners
-    if (this.listenForEvents) {
-      this._updateData = this._updateData.bind(this);
-      this._flushData = this._flushData.bind(this);
-      this.get('triEvents').on('update', this._updateData);
-      this.get('triEvents').on('flush', this._flushData);
-    }
-
-    // Fetch data once we have the `dataActions`
-    this.send('fetchData', this.get('dataActions'));
-  },
-
-  /**
-   * Remove the event listeners when the component is destroyed to prevent memory leaks
-   */
-  willDestroy() {
-    this._super(...arguments);
+    this.onFulfilledData = this.getWithDefault('onFulfilledData', noop);
```

Comment:
Since we only call this when data resolves we should rename this to avoid confusion. Users may be confused by `onFulfilledData` and think it gets called if a promise is resolved or rejected, where `onResolvedData` is clearly only called when a promise is resolved.